internal/app/singularity/cache_list_linux.go

```diff
@@ -31,82 +34,98 @@ func findSize(size int64) (string, error) {
 	return "", fmt.Errorf("failed to detect file size")
 }
 
-func listLibraryCache() error {
-	// loop through library cache
+// listLibraryCache will loop throught and list all library cache (~/.singularity/cache/library).
+// will return the amount of library containers, the total space thoughts containers are using,
+// and an error if one occures.
+func listLibraryCache(listFiles bool) (int, int64, error) {
+	var totalSize int64
+	count := 0
+
 	libraryCacheFiles, err := ioutil.ReadDir(cache.Library())
 	if err != nil {
-		return fmt.Errorf("unable to open library cache folder: %v", err)
+		return 0, 0, fmt.Errorf("unable to open library cache folder: %v", err)
 	}
 	for _, f := range libraryCacheFiles {
 		cont, err := ioutil.ReadDir(filepath.Join(cache.Library(), f.Name()))
```

Comment:
is this code trying to recursive list the contents of the cache? There's a function in the `filepath` package called `Walk`, that takes a directory and a function and it calls the function for each entry in the directory.

If that's not desired for some reason, this code should at least try to verify that f.Name() is a directory as expected, because ioutil.ReadDir will fail if you pass it a regular filename. You can use `os.Stat` to get a `os.FileInfo` structure and then call `IsDir` on that result to figure out if the entry is a directory before calling `ioutil.ReadDir`. In that way you can produce a warning about an unexpected file.